stack_configuration.go

```diff
@@ -64,3 +74,33 @@ func (s stackConfigurations) JSONSchemas(ctx context.Context, stackConfiguration
 
 	return raw.Bytes(), nil
 }
+
+// AwaitPrepared generates a channel that will receive the status of the stack configuration as it progresses.
+// The channel will be closed when the stack configuration reaches a status indicating that or an error occurs. The
+// read will be retried dependending on the configuration of the client. When the channel is closed,
+// the last value will either be a terminal status or an error.
+func (s stackConfigurations) AwaitPrepared(ctx context.Context, stackConfigurationID string) <-chan WaitForStatusResult {
```

Comment:
I'm curious what the advantage is to returning the read only channel as opposed to returning either the terminal status or error? The caller will inevitably have to write a range loop to read from it or am I missing some extra context here

Reply:
Oh wait I see the examples, it allows for easier handling within a `select`.